libmscore/scorefile.cpp

```diff
@@ -495,7 +495,9 @@ bool Score::saveCompressedFile(QFileInfo& info, bool onlySelection, bool createT
             MScore::lastError = tr("Open File\n%1\nfailed: %2").arg(info.filePath(), strerror(errno));
             return false;
             }
-      return saveCompressedFile(&fp, info, onlySelection, createThumbnail);
+      bool success = saveCompressedFile(&fp, info, onlySelection, createThumbnail);
+      fp.close();
```

Comment:
`QFile` destructor should close the file automatically, do we need to do that explicitly here?

Reply:
I believe we'd better do it where we still can (i.e. have the handle to it) rather than rely on someone else doing it for is (or not) and do it as soon as possible. "destructor should close" is a bit weak, isn't it?

Reply:
Hmm, https://doc.qt.io/qt-5/qfile.html#dtor.QFile so does that getting called when leaving the scope `fp` got instantiated in?

Reply:
If so almost none of the changes of this commit are needed, except for mscore/exportaudio.cpp  ?

Reply:
Yes, destructor is called automatically when this variable leaves the scope, so we can definitely rely on that.

Reply:
OK, so I'll amend that commit, later, probably tomorrow, just leaving the change to mscore/exportaudio.cpp  in

Reply:
Unless you merge it manually before that ;-)

Reply:
What the heck, I just did it now ;-)